graylog2-web-interface/src/components/events/fetchEvents.ts

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright (C) 2020 Graylog, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the Server Side Public License, version 1,
+ * as published by MongoDB, Inc.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * Server Side Public License for more details.
+ *
+ * You should have received a copy of the Server Side Public License
+ * along with this program. If not, see
+ * <http://www.mongodb.com/licensing/server-side-public-license>.
+ */
+
+import moment from 'moment';
+
+import * as URLUtils from 'util/URLUtils';
+import { adjustFormat } from 'util/DateTime';
+import type { SearchParams } from 'stores/PaginationTypes';
+import fetch from 'logic/rest/FetchProvider';
+import type { PaginatedResponse } from 'components/common/PaginatedEntityTable/useFetchEntities';
+import type { Event, EventDefinitionContext } from 'components/events/events/types';
+import { additionalAttributes } from 'components/events/Constants';
+import { extractRangeFromString } from 'components/common/EntityFilters/helpers/timeRange';
+import type { UrlQueryFilters } from 'components/common/EntityFilters/types';
+
+const url = URLUtils.qualifyUrl('/events/search');
+
+type FiltersResult = { filter: { alerts?: string }, timerange?: { from?: string, to?: string, type: string, range?: number}};
+
+const parseFilters = (filters: UrlQueryFilters) => {
+  const result: FiltersResult = { filter: {} };
+
+  if (filters.get('timestamp')?.[0]) {
+    const [from, to] = extractRangeFromString(filters.get('timestamp')[0]);
+
+    result.timerange = from
+      ? { from, to: to || adjustFormat(moment().utc(), 'internal'), type: 'absolute' }
+      : { type: 'relative', range: 0 };
+  } else {
+    result.timerange = { type: 'relative', range: 0 };
+  }
+
+  switch (filters?.get('alert')?.[0]) {
+    case 'true':
+      result.filter.alerts = 'only';
+      break;
+    case 'false':
+      result.filter.alerts = 'exclude';
+      break;
+    default:
+      result.filter.alerts = 'include';
+  }
+
+  return result;
+};
+
+const getConcatenatedQuery = (query: string, streamId: string) => {
+  if (!streamId) return query;
+
+  if (streamId && !query) return `source_streams:${streamId}`;
+
+  return `(${query}) AND source_streams:${streamId}`;
+};
+
+export const keyFn = (searchParams: SearchParams) => ['events', 'search', searchParams];
+
+export type EventsAdditionalData = {
+  context: {event_definitions?: EventDefinitionContext, streams?: EventDefinitionContext},
```

Comment:
If I am not mistaken this type is wrong. It reads like this is just one event definition while it is actually an object where the keys are the event definition ids and the values are the vent definitions.

Can you please check why this did not result in a TS error when using the context.

Reply:
I fixed it. Add EventDefinitionContexts. But I don't know why it's not throwing an error. It has to be somewhere in renderers here 
```
const eventDefinitionContext = eventsContext?.event_definitions?.[eventDefinitionId];
....
eventDefinitionContext.remediation_steps
```

Maybe because eventDefinitionId is a string the part `.[eventDefinitionId];` means any of the parameters from EventDefinitionContext